### ISSUE TITLE
Many reasons why a message with Datagram-flow-Id could be delayed

### ISSUE THREAD

**Issue by LPardue** [DavidSchinazi/draft-h3-datagram#12]
Section 6 makes the keen observation that:

```
   Since the QUIC STREAM frame that contains the "Datagram-Flow-Id"
   header could be lost or reordered, it is possible that an endpoint
   will receive an HTTP/3 datagram with a flow identifier that it does
   not know as it has not yet received the corresponding "Datagram-Flow-
   Id" header.  Endpoints MUST NOT treat that as an error; they MUST
   either silently discard the datagram or buffer it until they receive
   the "Datagram-Flow-Id" header.
```

This is accurate but, when thinking about HTTP/3 APIs, does not cover all possible reasons that might cause a request message to be processed after the corresponding DATAGRAM frame is available. For instance, the message might be blocked by QPACK, or the implementation might present DATAGRAMS before stream headers. 

We could boil the ocean with describing all possible reasons, so I think it might be better to shuffle the text to focus on describe the symptom (DATAGRAMS before HTTP messages) and the treatment. We can then give non-exhaustive example(s) as already done.

**Comment by DavidSchinazi** [comment]
I would prefer to not boil the ocean :)

I'll make an editorial change to make it clearer that this is just an example among others.
